src/modules/roc_fec/target_openfec/roc_fec/of_decoder.cpp

```diff
@@ -134,24 +139,44 @@ core::Slice<uint8_t> OFDecoder::repair(size_t index) {
     return buff_tab_[index];
 }
 
-void OFDecoder::reset() {
+void OFDecoder::end() {
     if (of_sess_ != NULL) {
         report_();
         destroy_session_();
     }
 
     reset_session_();
```

Comment:
I kept the original behavior when decoder resets a session after each `end` call.

Reply:
If I understand correctly, the original behaviour was to recreate the session exactly once per block, which is correct, and now it's possible that the session will be recreated twice, which is a bit worse.

Reply:
Well. We need to unify the responsibilities between `begin` and `end` because now it seems a little bit strange. Moreover, `decoder.end()` causes the session recreation but `encoder.end()` does not.

Reply:
Why is this important? Session (re)creation is an implementation detail hidden from the user. It can be created in begin, end, or even is set on demand -- as for me it doesn't change anything. On the other hand, recreating session twice is an early pessimization, and if I get it right it can be fixed trivially by removing this reset_session_() call. Isn't it?

Reply:
It is not important at all and you are right. The extra call to `reset_session_` has removed.